pkg/clients/handlers.go

```diff
@@ -0,0 +1,58 @@
+package clients
+
+import (
+	"net/http"
+
+	"github.com/canonical/identity-platform-admin-ui/internal/logging"
+	"github.com/canonical/identity-platform-admin-ui/internal/responses"
+	"github.com/go-chi/chi/v5"
+)
+
+type API struct {
+	service ServiceInterface
+
+	logger logging.LoggerInterface
+}
+
+func (a *API) RegisterEndpoints(mux *chi.Mux) {
+	mux.Get("/clients/{client_id}", a.GetClient)
+}
+
+func (a *API) WriteJSONResponse(w http.ResponseWriter, r *responses.Response) {
+	w.Header().Set("Content-Type", "application/json")
+	rr, err := r.PrepareResponse()
+	if err != nil {
+		a.logger.Errorf("Unexpected error: %s", err)
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+	w.Write(rr)
+	w.WriteHeader(r.Status)
+}
```

Comment:
i have some reserves in this, but i'd say crack on with the next 3 HTTP methods and see where u get with this and if it stays the same